test-cypress/integration/explore/visual.graph.spec.js

```diff
@@ -1,357 +1,203 @@
+    it('Test expand and collapse node info panel with single click', () => {
+        searchForResource(VALID_RESOURCE);
 
         // Click once on the node with the mouse to open node's info panel
-        cy.get('g[id="http://www.w3.org/TR/2003/PR-owl-guide-20031209/wine#' + validResource + '"] > circle')
-            .click({force: true})
-            .then(() => {
-                cy.get('pageslide > div.rdf-side-panel-content.break-word-alt.p-1.pt-2')
-                // Verify that a side panel is displayed containing info about the resource
-                    .should('be.visible', {timeout: 5000}).and('contain', validResource);
-            });
+        getTargetNode().click();
+        // Verify that a side panel is displayed containing info about the resource
+        getNodeInfoPanel().should('be.visible')
+            .find('.uri')
+            .should('be.visible')
+            .and('contain', VALID_RESOURCE);
+
+        // Close side panel and verify it's missing
+        getTargetNode().click();
+        getNodeInfoPanel().should('not.be.visible');
     });
 
-    it('Test remove child node', function () {
-        // Search for an valid resource
-        searchForResource(validResource);
-        // Verify redirection to existing visual graph
-        cy.url({timeout: 5000})
-            .should('include', validResource);
-
+    it('Test remove child node', () => {
+        searchForResource(VALID_RESOURCE);
         // Verify that before given node is removed there are 3 of them
-        cy.get('g.nodes-container > g.node-wrapper').should('have.length', '3');
-
+        getNodes().should('have.length', 3);
         // Click once on node different than parent one with the mouse
-        cy.get('g.nodes-container > g.node-wrapper > circle', {waitForAnimations: true, animationDistanceThreshold: 0}).eq(1)
+        cy.get('.node-wrapper circle').eq(1)
         // timeout is needed because mouseover event will result in
         // pop-up of menu icons only if nodes are not moving
-            .should('be.visible', {timeout: 5000}).wait(5000)
-            .trigger('mouseover').wait(1000)
-            .then(() => {
-                // Select remove function
-                cy.get('g.nodes-container > g.menu-events > g.close-icon > circle')
-                    .trigger('click', {force: true})
-                    .then(() => {
-                        // Verify that link between parent node and the one child node is expanded
-                        cy.get('g.nodes-container > g.link-wrapper')
-                            .should('have.length', '1');
-                        // Verify that the USRegion node is not the only node left in the graph
-                        cy.get('g.nodes-container > g.node-wrapper')
-                            .its('length')
-                            .should('to.be', '2');
-                    });
-            });
+            .should('be.visible').wait(5000)
```

Comment:
This wait has no purpose here ? There is no command in the chain

Reply:
The comment explains why it's needed and in fact it really is.

Reply:
I see, maybe tuning https://docs.cypress.io/guides/core-concepts/interacting-with-elements.html#Animations for this spec could result in more stable tests and lack of waits.